.github/workflows/test.yaml

```diff
@@ -11,10 +11,22 @@ jobs:
       - name: Download dependencies
         run: go mod tidy
 
-      - name: Run Unit Tests
+      - name: Run unit tests
         run: |
           make test
           bash -c '/bin/bash <(curl -s https://codecov.io/bash)'
 
       - name: Build executable
         run: go build -v ./...
+
+  integration:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Start services
+        run: docker-compose up -d --build elasticsearch server
+      - name: Run integration tests
+        run: make integration
+      - name: Teardown services
+        if: ${{ always() }}
```

Comment:
what's the reason for this?

Reply:
For the overall teardown, I'm not familiar enough yet with github actions if it is safe to just leave docker-compose running. There is a general cleanup task the runs with the output of killing processes, but I don't know how projects/workers/etc. are reused.

The `if: ${{ always()}}` is so the the teardown step will run whether or not any other previous steps fail.